### PR TITLE
TNT-40658 - updated openapi models using customized resttemplate java generator

### PR DESCRIPTION
**Pull request by dcottingham**
These models were created using a customized openapi java generator, along with customized templates.  In order to automate model generation I will need to fork this repo and add my custom java generator code since all customizations could not be made via templates alone:  https://github.com/OpenAPITools/openapi-generator

### CODE REVIEW

File: src/main/java/com/adobe/target/delivery/v1/model/MboxResponse.java [hunk 1]
```diff
@@ -13,13 +13,18 @@
...
-import java.util.*;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
 
 /** Mbox response object. */
-@JsonInclude(JsonInclude.Include.NON_NULL)
 public class MboxResponse {
```
Comment (artur-ciocanu) [inline src/main/java/com/adobe/target/delivery/v1/model/MboxResponse.java:24]: @dcottingham please make sure that this is expected. I am not sure why we have `$_type`.

File: src/main/java/com/adobe/target/delivery/v1/model/RequestDetails.java [hunk 2]
```diff
@@ -22,10 +21,12 @@
 /**
  * Object common for prefetch, execute and notifications in order to specify the request details.
  */
-@JsonInclude(JsonInclude.Include.NON_NULL)
 public class RequestDetails {
```
Comment (artur-ciocanu) [inline src/main/java/com/adobe/target/delivery/v1/model/RequestDetails.java:24]: Same here, it's not clear why we need `$_type`.

File: src/test/java/com/adobe/target/edge/client/entities/TargetDeliveryLocalGeoTest.java [hunk 3]
```diff
@@ -24,23 +24,11 @@
 import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.verify;
```
Comment (artur-ciocanu) [inline src/test/java/com/adobe/target/edge/client/entities/TargetDeliveryLocalGeoTest.java:26]: You should probably configure your IDE to do not collapse imports.

File: src/test/java/com/adobe/target/edge/client/entities/TargetDeliveryLocalPropertyTest.java [hunk 4]
```diff
@@ -22,23 +22,11 @@
 import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.verify;
```
Comment (artur-ciocanu) [inline src/test/java/com/adobe/target/edge/client/entities/TargetDeliveryLocalPropertyTest.java:24]: Same here

**Review by artur-ciocanu — CHANGES_REQUESTED** [review]
@dcottingham I have left some comments, overall the changes are great. However we have to be careful around moving currently exposed classes to other places.

Ideally we should have tests that would ensure that classes available on `com.adobe.target.edge.client.model` are available and working as expected. Otherwise if customers will upgrade and we moved something to a different package, their code might fail.

**Review by jasonwaters — APPROVED** [review]
LGTM!

**Pull request merged**